### PR TITLE
update eof for windows

### PR DESCRIPTION
**Pull request by nihardamar**
On windows it's not correctly matching the EOF that's received from content_shell.
In this change we update to use the correct format to end the process correctly.

### CODE REVIEW

File: tools/wptrunner/wptrunner/executors/executorcontentshell.py [hunk 1]
```diff
@@ -43,6 +44,9 @@ class ContentShellTestPart(ProtocolPart):
     name = "content_shell_test"
     eof_marker = "#EOF" + linesep  # Marker sent by content_shell after blocks.
 
+    if sys.platform == "win32":
+        eof_marker = (b'#EOF\n').decode()
```
Comment (foolip) [inline tools/wptrunner/wptrunner/executors/executorcontentshell.py:48]: Would this be equivalent to this string literal?

```suggestion
        eof_marker = '#EOF\n'
```

Comment (nihardamar) [reply tools/wptrunner/wptrunner/executors/executorcontentshell.py:48]: I think so, let me test this using the literal string, 
reason I used decode was to try to closely match what our old system "run_web_tests" was using.

Here's a doc with the issue + comparison with old system
https://docs.google.com/document/d/1lpVTWOtIiUrQVxY9CUb5RgJIK_O0XmjZTga89grk2X4/edit?resourcekey=0-ff5CbM-cdqKTIUIwvIj8FQ

Comment (foolip) [reply tools/wptrunner/wptrunner/executors/executorcontentshell.py:48]: Code like that might have been for Python 2/3 compat at some point, but now it should be equivalent.

Comment (nihardamar) [reply tools/wptrunner/wptrunner/executors/executorcontentshell.py]: ok cool, just checked results from CL and it looks good, I updated to use plain text.
https://chromium-review.googlesource.com/c/chromium/src/+/4020962

File: tools/wptrunner/wptrunner/executors/executorcontentshell.py [hunk 2]
```diff
@@ -43,6 +44,9 @@ class ContentShellTestPart(ProtocolPart):
     name = "content_shell_test"
     eof_marker = "#EOF" + linesep  # Marker sent by content_shell after blocks.
```
Comment (jonathan-j-lee) [inline tools/wptrunner/wptrunner/executors/executorcontentshell.py:45]: Drive-by: What's stopping us from assigning `"#EOF\n"` always?

Using platform-dependent `linesep` seems [wrong](https://source.chromium.org/chromium/chromium/src/+/main:docs/how_to_extend_web_test_framework.md;l=245;drc=81f891fd023c66720731b86b7f280dae2311b688;bpv=0;bpt=0).

File: tools/wptrunner/wptrunner/executors/executorcontentshell.py [hunk 3]
```diff
@@ -95,6 +94,10 @@ def _read_text_block(self, deadline=None):
             if line.endswith(self.eof_marker):
                 result += line[:-len(self.eof_marker)]
                 break
+            elif line.endswith('#EOF\r\n'):
+                result += line
+                result += 'Got a CRLF-terminated #EOF - this is a driver bug.'
```
Comment (WeizhongX) [inline tools/wptrunner/wptrunner/executors/executorcontentshell.py:99]: This should be:

result += line[:-len('#EOF\r\n')]
logger.warning('Got a CRLF-terminated #EOF - this is a driver bug.')

Comment (nihardamar) [reply tools/wptrunner/wptrunner/executors/executorcontentshell.py:99]: I was thinking here to have the result log the full line so a developer can see what is being sent, or do you think it's fine to leave this out as it could be implied what is being sent?

File: tools/wptrunner/wptrunner/executors/executorcontentshell.py [hunk 4]
```diff
@@ -67,7 +66,7 @@ def do_test(self, command, timeout=None):
     def _send_command(self, command):
         """Sends a single `command`, i.e. a URL to open, to content_shell.
         """
-        self.stdin_queue.put((command + linesep).encode("utf-8"))
+        self.stdin_queue.put((command + "\n").encode("utf-8"))
```
[the remark this answers is not in the document]
Comment (nihardamar) [reply tools/wptrunner/wptrunner/executors/executorcontentshell.py:69]: Running a test on the CL now with the updated changes.

File: tools/wptrunner/wptrunner/executors/executorcontentshell.py (hunk 3 above)
Comment (jonathan-j-lee) [reply tools/wptrunner/wptrunner/executors/executorcontentshell.py:99]: Personally, I'm of the opinion that it's not worth having this special case for `#EOF\r\n`. AFAICT, `content_shell` [does not return `\r`](https://source.chromium.org/chromium/chromium/src/+/main:content/web_test/browser/test_info_extractor.h;l=44;drc=4e1b7bc33d42b401d7d9ad1dcba72883add3e2af;bpv=0;bpt=0). If wptrunner is inserting `\r` before it goes into `stdout_queue`, then we need to fix that bug instead of dealing with the alternative marker downstream here.

Comment (WeizhongX) [reply tools/wptrunner/wptrunner/executors/executorcontentshell.py:99]: https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/tools/blinkpy/web_tests/port/driver.py;drc=4d3216caa7ad9b2dbf00316c6cfa6d76fd012edb;l=693

Content shell has the code to handle '\r\n'. Doing this the same way as rwt could save us some headache.

**Comment by nihardamar** [comment]
Tested new changes in CL below:
https://chromium-review.googlesource.com/c/chromium/src/+/4020962?tab=checks

File: tools/wptrunner/wptrunner/executors/executorcontentshell.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (nihardamar) [reply tools/wptrunner/wptrunner/executors/executorcontentshell.py:69]: https://chromium-review.googlesource.com/c/chromium/src/+/4020962?tab=checks

**Review by WeizhongX — APPROVED** [review]
